### PR TITLE
PWA notifications for Connect app

### PR DESCRIPTION
**Pull request by valentinludu**
How it works:
1. The user needs to install the pwa app and login
    - This is possible with the help of the manifest file (manifest.json)
2. The user needs to accept notifications from the popup screen
     - Each time a user didn't set his permissions we ask permission to send him notifications (layout).
     - Usually this happens only once
2. We register a subscription for a certain user from a certain device in our db 
    - Once the user accepts the notifications we are saving this subscription permission in our db with the help of a server action (actionSaveSubscription)
4. Once a day at 10 AM a cron job runs and sends a notification to users 
    - Made possible with the web-push package (cron task)
6. The service worker file receives the notification from the backend and serves that notification to the user's device
    - sw.js file handles it

### CODE REVIEW

File: apps/connect/app/api/push-notifications/route.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { prisma } from '@charmverse/core/prisma-client';
```
Comment (motechFR) [inline apps/connect/app/api/push-notifications/route.ts:1]: We should delete this file, or make it a dev only route

File: apps/connect/app/layout.tsx [hunk 2]
```diff
@@ -1,28 +1,68 @@
...
   themeColor: '#fff'
 };
 
-export default function RootLayout({
+export default async function RootLayout({
   children
 }: Readonly<{
   children: ReactNode;
 }>) {
+  const session = await getIronSession<SessionData>(cookies(), getIronOptions());
```
Comment (motechFR) [inline apps/connect/app/layout.tsx:61]: small suggestion, could we wrap this in a util method called getSession() to make it easier to consume?

File: apps/connect/app/api/push-notifications/route.ts (hunk 1 above)
Comment (valentinludu) [reply apps/connect/app/api/push-notifications/route.ts:1]: Done

File: apps/connect/app/layout.tsx (hunk 2 above)
Comment (valentinludu) [reply apps/connect/app/layout.tsx:61]: Done

**Review by motechFR — APPROVED** [review]

File: apps/connect/app/layout.tsx (hunk 2 above)
Comment (valentinludu) [reply apps/connect/app/layout.tsx:61]: I finally made it to work. Session should have the same cookieName as the one from connect-api

**Pull request merged**